src/lib/util/zippath.cpp

```diff
@@ -140,13 +140,13 @@ static void parse_parent_path(const char *path, int *beginpos, int *endpos)
 //  zippath_parent - retrieves the parent directory
 // -------------------------------------------------
 
-std::string &zippath_parent(std::string &dst, const char *path)
+std::string &zippath_parent(std::string &dst, const std::string &path)
 {
-	int pos;
+	std::string::size_type pos;
 	parse_parent_path(path, &pos, nullptr);
 
 	if (pos >= 0)
```

Comment:
With `pos` changed to `std::string::size_type` this comparison is always true due to limited range of type. You can't use a negative value as a sentinel.